runtime/common/src/crowdloan/migration.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2017-2020 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::*;
+use frame_support::generate_storage_alias;
+
+/// Migrations for using fund index to create fund accounts instead of para ID.
+pub mod crowdloan_index_migration {
+	use super::*;
+
+	// The old way we generated fund accounts.
+	fn old_fund_account_id<T: Config>(index: ParaId) -> T::AccountId {
```

Comment:
is the only difference between this and the one the new one that this accepts `ParaId`, and the new one accepts `FundIndex`? If so can't we just cast `ParaId as FundIndex` and avoid this almost duplicate impl?

Reply:
nope, i dont like what you are suggesting at all. These are type safe functions and the whole point of impling this a second time here is that this was the ORIGINAL implementation, and what I am using to be the basis of my migration here.